### DISCUSSION TITLE
Timeline of beats and accuracy when watching replay

### DISCUSSION THREAD

**Discussion by bluelightzero** [ppy/osu#17885] [Ideas]
It would be nice to see clearly at a glance which parts of a beat I struggle with so I know pattern types I suck at.

It would also be nice to see a snapshot of this failure so you can see what you did wrong at a glance after you lose.

----

I propose that a scrolling graph like this gets added when watching demos: 

![image](https://user-images.githubusercontent.com/4492982/164027837-89ce11d4-b73b-4049-bf97-1c3275b701df.png)

The vertical coloured bars represent the beats you needed to hit.

The white dots represent the inputs you made. 

The higher a white dot is, the further away from the current circle's center your aim was

Dots above the beat lines are ones that would count as a miss. (You can see this in the red combo)

Dots that don't line up with beat lines are ones that are timed incorrectly (You can see this in the green combo, the input was too soon)

**Comment by Theighlin** [comment]
Related to #5681?

**Reply by bluelightzero** [reply]
Kind of. This is more about showing *how* you went wrong than just showing that you went wrong.
